src/cargo/util/toml/mod.rs

```diff
@@ -1513,6 +1514,43 @@ impl TomlManifest {
     }
 }
 
+/// Returns the name of the README file for a `TomlProject`.
+fn readme_for_project(package_root: &Path, project: &Box<TomlProject>) -> Option<String> {
+    match &project.readme {
+        None => default_readme_from_package_root(package_root),
+        Some(value) => match value.as_str() {
+            "false" => None,
+            _ => Some(value.clone()),
+        },
+    }
+}
+
+const DEFAULT_README_FILES: [&str; 3] = ["README.md", "README.txt", "README"];
+
+/// Checks if a file with any of the default README file names exists in the package root.
+/// If so, returns a `String` representing that name.
+fn default_readme_from_package_root(package_root: &Path) -> Option<String> {
+    _default_readme_from_package_root(package_root).ok()
+}
+
+fn _default_readme_from_package_root(package_root: &Path) -> CargoResult<String> {
+    for entry in package_root.read_dir()? {
```

Comment:
Instead if reading for files and looking for filenames, could this perhaps just test if any of the default readme files exist as a file? (e.g. `package_root.join(candidate).is_file()`)